filament/src/Material.cpp

```diff
@@ -45,10 +45,14 @@ using namespace driver;
 
 
 struct Material::BuilderDetails {
-    const void* mPayload = nullptr;
-    size_t mSize = 0;
     filaflat::MaterialParser* mMaterialParser = nullptr;
     bool mDefaultMaterial = false;
+
+    // We have to use the heap to stash the buffer descriptor because BuilderBase uses copy
+    // assignment on the details instance, and BufferDescriptor has a deleted copy constructor. For
+    // the same reason, we also need to use shared_ptr rather than unique_ptr.
+    // TODO: consider refactoring BuilderBase to allow descendants that do not support copying.
+    std::shared_ptr<BufferDescriptor> mBufferDescriptor = std::make_shared<BufferDescriptor>();
```

Comment:
You could have a copy-ctor on BuilderDetails instead, e.g.:
```
BuilderDetails::BuilderDetails(BuilderDetails const& rhs) 
    : mBufferDescriptor(rhs.data, rhs.size, rhs.callback) {
}
```

You may have to make BufferDescriptor::callback public or add a getCallback().